codegen/smithy-go-codegen/src/test/java/software/amazon/smithy/go/codegen/SemanticVersionTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.Comparator;
+import java.util.Optional;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+public class SemanticVersionTest {
+
+    @Test
+    public void testSemanticVersion() {
+        SemanticVersion version = SemanticVersion.parseVersion("1.2.3");
+        assertThat(version.toString(), Matchers.equalTo("1.2.3"));
+    }
```

Comment:
Would be nice to have test that uses `gohash` format, and also test case for invalid/incomplete version